lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -1109,6 +1109,17 @@ void PatchEntryPointMutate::addSpecialUserDataArgs(SmallVectorImpl<UserDataArg>
       auto numWorkgroupsPtrTy = PointerType::get(FixedVectorType::get(builder.getInt32Ty(), 3), ADDR_SPACE_CONST);
       userDataArgs.push_back(UserDataArg(numWorkgroupsPtrTy, "numWorkgroupsPtr", UserDataMapping::Workgroup, nullptr));
     }
+  } else if (m_shaderStage == ShaderStageTask) {
+    auto taskIntfData = m_pipelineState->getShaderInterfaceData(ShaderStageTask);
+    specialUserDataArgs.push_back(UserDataArg(FixedVectorType::get(builder.getInt32Ty(), 3), "meshTaskDispatchDims",
+                                              UserDataMapping::MeshTaskDispatchDims,
+                                              &taskIntfData->entryArgIdxs.task.dispatchDims));
+    specialUserDataArgs.push_back(UserDataArg(builder.getInt32Ty(), "meshTaskRingIndex",
+                                              UserDataMapping::MeshTaskRingIndex,
+                                              &taskIntfData->entryArgIdxs.task.baseRingEntryIndex));
+    specialUserDataArgs.push_back(UserDataArg(builder.getInt32Ty(), "meshPipeStatsBuf",
+                                              UserDataMapping::MeshPipeStatsBuf,
+                                              &taskIntfData->entryArgIdxs.task.pipeStatsBuf));
```

Comment:
Could we add a test that will exercise this code?  You could create an LGC lit test and check that after entry point mutation the parameters are correct.  If we could check that their use is correct as well that would be even better.

Reply:
Yes, I will try to add an empty task shader to test the entry-point arguments.